microcosm_metrics/decorators.py

```diff
@@ -0,0 +1,36 @@
+"""
+Metrics decorators.
+
+"""
+from functools import wraps
+from time import time
+
+from microcosm_metrics.naming import name_for
+
+
+def configure_metrics_timing(graph):
+    """
+    Configure a timing decorator.
+
+    """
+    def metrics_timing(name):
+        """
+        Create a decorator that times a specific context.
+
+        """
+        def decorator(func):
+            @wraps(func)
+            def wrapper(*args, **kwargs):
+                start_time = time()
+                try:
+                    return func(*args, **kwargs)
+                finally:
+                    end_time = time()
+                    # NB: the statsd client doesn't actually have this interface
```

Comment:
At the moment, I've only tested with datadog. I'd like to maintain generalized support (the only diff AFAIK is that datadog supports passing tags with metrics and statsd does not). I'm pretty sure I'll need to change something here before "true" statsd works.